File: solutions/004/004.js
function isPalindrome(toCheck) {
  return String(toCheck).split('').reverse().join('') === String(toCheck)
}

function largestPalindromeProduct(last, first) {
  let largest = 0
  for (let a = last; a >= first; a--)
    for (let b = last; b >= first; b--) {
      const calculated = a * b
      if (calculated > largest && isPalindrome(calculated))
        largest = calculated
    }
  return largest
}

console.log(largestPalindromeProduct(999, 100))

File: solutions/004/_004.java
public class _004 {
    public static void main(String[] args) {
        System.out.println(largestPalindromeProduct(999, 100));
    }

    public static boolean isPalindrome(int number) {
        int n = number;
        int reverse = 0;
        while (n > 0) {
            reverse = reverse * 10 + n % 10;
            n /= 10;
        }
        return number == reverse;
    }

    public static int largestPalindromeProduct(int from, int to) {
        int largest = 0;
        for (int a = from; a >= to; a--)
            for (int b = from; b >= to; b--) {
                int calculated = a*b;
                if (calculated > largest && isPalindrome(calculated))
                    largest = calculated;
            }
        return largest;
    }
}
